### PR TITLE
WIP - MARVEL-2100 - Add logging to `registrator`

### PR DESCRIPTION
**Pull request by glitch100**
## Overview

As per ticket MARVEL-2100 we have added additional logging, stopped ignoring some errors and added a global recover for any panics that somehow went under the radar.

## Contents
- `docker-compose.yml` file for easier debugging on Windows
- `recover()` deferred function added
- Additional logging
- Some ignored errors reinstated

## External related issues
Is it worth us upgrading to the official docker go SDK instead of third party? It may alleviate our issues.

### CODE REVIEW

File: bridge/bridge.go [hunk 1]
```diff
@@ -461,5 +471,10 @@ var Hostname string
 func init() {
 	// It's ok for Hostname to ultimately be an empty string
 	// An empty string will fall back to trying to make a best guess
-	Hostname, _ = os.Hostname()
+	hostname, err := os.Hostname()
```
Comment (chrisgilbert) [inline bridge/bridge.go]: I think an error here might be OK, given the comment?

File: registrator.go [hunk 2]
```diff
@@ -208,12 +217,11 @@ func main() {
 	for msg := range events {
 		switch msg.Status {
```
Comment (chrisgilbert) [inline registrator.go]: It looks like the docker API changed back on v1.22.  The client we are using seems to account for that and try to provide something compatible, but the `Status` string seems to have gone from docker itself.  I wonder if these API changes could be causing any of our problems? https://github.com/fsouza/go-dockerclient/blob/a107859ba6dd3406069157ed63a31ba859f73228/event.go#L40

File: registrator.go [hunk 3]
```diff
@@ -60,6 +60,15 @@ func main() {
 	logging.Configure()
 
 	log.Infof("Starting registrator %s ...", Version)
+	quit := make(chan struct{})
+	defer func() {
```
Comment (chrisgilbert) [inline registrator.go]: Cool, I bet this will do the trick!

File: registrator.go (hunk 2 above)
Comment (glitch100) [reply registrator.go]: Is this something we should test out do you reckon?  It must work to some degree otherwise nothing would be registered

Comment (glitch100) [reply registrator.go]: O my, I started moving it across last night. Glad I didn't finish it aha. I think it's worth moving to an official library over a 3rd party in this scenario, but see what Jamie says

**Pull request merged**